torch/_inductor/codegen/rocm/ck_conv_template.py

```diff
@@ -0,0 +1,556 @@
+# mypy: allow-untyped-defs
+import copy
+import logging
+import random
+from typing import Tuple
+
+
+try:
+    import ck4inductor  # type: ignore[import]
+except ImportError:
+    ck4inductor = None
+
+if ck4inductor is not None:
+    from ck4inductor.grouped_conv_fwd.gen_instances import (  # type: ignore[import]
+        gen_conv_ops_library,
+    )
+    from ck4inductor.grouped_conv_fwd.op import (  # type: ignore[import]  # noqa: TCH002
+        CKGroupedConvFwdOp,
+    )
+else:
+
+    def gen_conv_ops_library():
+        return []
+
+
+from torch._inductor import config
+from torch._inductor.codegen.rocm.ck_template import CKTemplate
+from torch._inductor.codegen.rocm.rocm_kernel import ROCmTemplateKernel
+from torch._inductor.utils import IndentedBuffer
+
+
+log = logging.getLogger(__name__)
+
+
+def torch_layout_to_ck_layouts(torch_layout):
+    # logically, torch tensors are always NCHW,
+    # and channels-last memory layout is visible in the strides
+    if torch_layout.stride[-1] == 1:
```

Comment:
we can use `statically_known_equals(stride[-1], 1)` here for dynamic shape support

https://github.com/pytorch/pytorch/blob/134f6cda7e192046921f44b9565a3b56cd28158d/torch/_inductor/sizevars.py#L340C9-L340C32